### PR TITLE
docs(blockchain-scanner): Update documentation for the new `zebra-scanner` binary

### LINKED ISSUE #8586: Documentation changes for the new `zebra-scanner` binary
After we remove the scanning functionality from zebrad (https://github.com/ZcashFoundation/zebra/issues/8582) some of the current documentation will need to change for example: https://zebra.zfnd.org/user/shielded-scan-grpc-server.html

All documentation changes and additions related to running the blockchain scanner as a standalone process can be done as part of this issue.

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

After we added the `zebra-scanner` binary the documentation we had to scan went out of date. 

Close https://github.com/ZcashFoundation/zebra/issues/8586

## Solution

Upgrade documentation.

### PR Author's Checklist

- [x] The PR name will make sense to users.
- [ ] The PR provides a CHANGELOG summary.
- [ ] The solution is tested.
- [ ] The documentation is up to date.
- [ ] The PR has a priority label.

### PR Reviewer's Checklist

- [ ] The PR Author's checklist is complete.
- [ ] The PR resolves the issue.

### CODE REVIEW

File: book/src/user/shielded-scan-grpc-server.md [hunk 1]
```diff
@@ -4,14 +4,12 @@
 
 ### Setup
 
-After setting up [Zebra Shielded Scanning](https://zebra.zfnd.org/user/shielded-scan.html), add a `listen_addr` field to the shielded-scan configuration:
+After setting up [Zebra Shielded Scanning](https://zebra.zfnd.org/user/shielded-scan.html), adyou can add a `listen-addr` argument to the scanner binary:
```
Comment (arya2) [inline book/src/user/shielded-scan-grpc-server.md]:
```suggestion
After setting up [Zebra Shielded Scanning](https://zebra.zfnd.org/user/shielded-scan.html), you can add a `listen-addr` argument to the scanner binary:
```

File: book/src/user/shielded-scan-grpc-server.md [hunk 2]
```diff
@@ -4,14 +4,12 @@
...
 
-```toml
-[shielded_scan]
-listen_addr = "127.0.0.1:8231"
-```
 
-Then, run `zebrad` to start the scan gRPC server.
+```bash
+RUST_LOG=info zebra-scanner --sapling-keys-to-scan '{"key":"zxviews1q0duytgcqqqqpqre26wkl45gvwwwd706xw608hucmvfalr759ejwf7qshjf5r9aa7323zulvz6plhttp5mltqcgs9t039cx2d09mgq05ts63n8u35hyv6h9nc9ctqqtue2u7cer2mqegunuulq2luhq3ywjcz35yyljewa4mgkgjzyfwh6fr6jd0dzd44ghk0nxdv2hnv4j5nxfwv24rwdmgllhe0p8568sgqt9ckt02v2kxf5ahtql6s0ltjpkckw8gtymxtxuu9gcr0swvz", "birthday_height": 419200}' --zebrad-cache-dir /media/alfredo/stuff/chain/zebra --zebra-rpc-listen-addr '127.0.0.1:8232' --listen-addr '127.0.0.1:8231'
```
Comment (arya2) [inline book/src/user/shielded-scan-grpc-server.md]: I think the `RUST_LOG=info` should be unnecessary here, abscissa is setting it as "info" by default in zebrad, we could do the same here?

I'll leave a suggestion below in the `scanner` bin to do this.

File: zebra-scan/src/bin/scanner/main.rs [hunk 3]
```diff
@@ -43,6 +43,8 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let args = Args::from_args();
```
Comment (arya2) [inline zebra-scan/src/bin/scanner/main.rs:48]: If we want to use info level logging by default, we could replace the `tracing_subscriber::EnvFilter::from_default_env()` call just above here with:

```rust
    let tracing_filter: String = match std::env::var("RUST_LOG") {
        Ok(val) if !val.is_empty() => val,
        _ => "info".to_string(),
    };

    tracing_subscriber::fmt::fmt()
        .with_env_filter(tracing_filter)
```

File: book/src/user/shielded-scan.md [hunk 4]
```diff
@@ -1,90 +1,75 @@
 # Zebra Shielded Scanning
 
-This document describes Zebra's shielded scanning from users' perspective.
+The `zebra-scanner` binary is a standalone application that utilizes Zebra libraries to scan for transactions associated with specific Sapling viewing keys. It stores the discovered transactions and scanning progress data in a RocksDB database.
 
+For this application to function, it requires a `zebrad` node to be running concurrently with the scanner application.
```
Comment (arya2) [inline book/src/user/shielded-scan.md]: Nitpick/Optional: Could we also mention that it needs access to Zebra's state cache?

```suggestion
For this application to function, it requires access to a Zebra node's RPC server and state cache.
```

File: book/src/user/shielded-scan.md [hunk 5]
```diff
@@ -1,90 +1,75 @@
...
+ 
 For now, we only support Sapling, and only store transaction IDs in the scanner results database.
-Ongoing development is tracked in issue [#7728](https://github.com/ZcashFoundation/zebra/issues/7728).
+
+Ongoing development is tracked in issue [#7728](https://github.com/ZcashFoundation/zebra/issues/7728) and [#8573](https://github.com/ZcashFoundation/zebra/issues/8573).
```
Comment (arya2) [inline book/src/user/shielded-scan.md]: Optional: It looks like #8573 is almost done if we want to omit it here.
```suggestion
Ongoing development is tracked in issue [#7728](https://github.com/ZcashFoundation/zebra/issues/7728).
```

File: book/src/user/shielded-scan.md [hunk 6]
```diff
@@ -1,90 +1,75 @@
...
 
 ## Important Security Warning
 
 Zebra's shielded scanning feature has known security issues. It is for experimental use only.
 
-Do not use regular or sensitive viewing keys with Zebra's experimental scanning
-feature. Do not use this feature on a shared machine. We suggest generating new
-keys for experimental use or publicly known keys.
+Do not use regular or sensitive viewing keys with Zebra's experimental scanning feature. Do not use this feature on a shared machine. We suggest generating new keys for experimental use or publicly known keys.
```
Comment (arya2) [inline book/src/user/shielded-scan.md]:
```suggestion
Do not use regular or sensitive viewing keys with Zebra's experimental scanning feature. Do not use this feature on a shared machine. We suggest generating new keys for experimental use or using publicly known keys.
```

File: book/src/user/shielded-scan.md [hunk 7]
```diff
@@ -1,90 +1,75 @@
...
-or greater, or the `main` branch to get the latest features, and enable the
-`shielded-scan` feature during the build. You can also use Rust's `cargo` to
-install the latest release:
+Use [Zebra 1.9.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.9.0) or greater, or the `main` branch to get the latest features.
+
+You can also use Rust's `cargo` to install `zebra-scanner` from the latest release Zebra repository:
 
 ```bash
-cargo install --features shielded-scan --locked --git https://github.com/ZcashFoundation/zebra zebrad
+cargo install --locked --git https://github.com/ZcashFoundation/zebra zebra-scan
```
Comment (arya2) [inline book/src/user/shielded-scan.md:24]: This probably also installs `zebrad` (as `zebrad-for-scanner`), we may want to hide that behind a feature later.

File: book/src/user/shielded-scan.md [hunk 8]
```diff
@@ -1,90 +1,75 @@
...
-- the `[shielded_scan]` table for database settings, and
-- the `[shielded_scan.sapling_keys_to_scan]` table for diversifiable full viewing keys.
-
-Sapling diversifiable/extended full viewing keys strings start with `zxviews` as
-described in
-[ZIP-32](https://zips.z.cash/zip-0032#sapling-extended-full-viewing-keys).
-
-For example, to scan the block chain with the [public ZECpages viewing
-key](https://zecpages.com/boardinfo), use:
+Before starting, ensure a `zebrad` node is running locally with the RPC endpoint open. Refer to the [lightwalletd zebrad setup](https://zebra.zfnd.org/user/lightwalletd.html#configure-zebra-for-lightwalletd) or [zebrad mining setup](https://zebra.zfnd.org/user/mining.html#configure-zebra-for-mining) for instructions.
```
Comment (arya2) [inline book/src/user/shielded-scan.md:39]: Optional: We could be more specific about what's needed here too, but "is running locally" also seems fine, hopefully this won't be a constraint in the future anyway.

File: book/src/user/shielded-scan.md [hunk 9]
```diff
@@ -1,90 +1,75 @@
...
-- birthday lower than the Sapling activation height defaults to Sapling activation height.
-- birthday greater or equal than Sapling activation height will start scanning at provided height, improving scanner speed.
+- A zebrad cache state directory. This can be obtained from the running zebrad configuration file, under the `state` section in the `cache_dir` field.
+- A key to scan with, optionally including a birthday height, which specifies the starting height for the scanning process for that key.
+- A zebrad RPC endpoint address. This can be found in the running zebrad configuration file, under the `rpc` section in the `listen_addr` field.
 
-## Scanning the Block Chain
 
-Simply run
+Pass the data as arguments, for example:
```
Comment (arya2) [inline book/src/user/shielded-scan.md]:
```suggestion
Sapling diversifiable/extended full viewing keys strings start with `zxviews` as
described in
[ZIP-32](https://zips.z.cash/zip-0032#sapling-extended-full-viewing-keys).

For example, to scan the block chain with the [public ZECpages viewing
key](https://zecpages.com/boardinfo), use:
```

File: zebra-scan/src/bin/scanner/main.rs [hunk 10]
```diff
@@ -142,3 +144,19 @@ pub struct Args {
...
+fn validate_dir(dir: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    match dir.try_exists() {
+        Ok(true) => Ok(()),
+        Ok(false) => {
+            let err_msg = format!("Directory {} does not exist.", dir.display());
+            error!("{}", err_msg);
+            Err(std::io::Error::new(std::io::ErrorKind::NotFound, err_msg).into())
+        }
+        Err(e) => {
+            error!("Directory {} does not exist: {:?}", dir.display(), e);
```
Comment (arya2) [inline zebra-scan/src/bin/scanner/main.rs]:
```suggestion
            let err_msg = format!("directory {} does not exist.", dir.display());
            error!("{}", err_msg);
            Err(std::io::Error::new(std::io::ErrorKind::NotFound, err_msg).into())
        }
        Err(e) => {
            error!("directory {} could not be accessed: {:?}", dir.display(), e);
```

**Review by arya2 — COMMENTED** [review]
This is looking good!

File: zebra-scan/src/bin/scanner/main.rs (hunk 3 above)
Comment (oxarbitrage) [reply zebra-scan/src/bin/scanner/main.rs:48]: done in https://github.com/ZcashFoundation/zebra/pull/8675/commits/510ac5ed2c9a699aacc08f72093d98414a9ed9c4

File: book/src/user/shielded-scan-grpc-server.md (hunk 2 above)
Comment (oxarbitrage) [reply book/src/user/shielded-scan-grpc-server.md]: I think you are right and this is better. Did the change to the code at https://github.com/ZcashFoundation/zebra/pull/8675/commits/510ac5ed2c9a699aacc08f72093d98414a9ed9c4 and to the doc in https://github.com/ZcashFoundation/zebra/pull/8675/commits/ea74b744bc1b64eea54eb936e47d591aa12d02c2

**Review by arya2 — APPROVED** [review]
Looks good, thank you!

**Pull request merged**